### PR TITLE
Add function of reviewing flashcards

### LINKED ISSUE #8: Review flashcards
As a user I can review the cards that are due today so that I can remember them better

### PR DESCRIPTION
**Pull request by kdh3799**

### CODE REVIEW

**Review by joulev — COMMENTED** [review]
Could we get some tests as well?

File: src/main/java/com/clanki/Clanki.java [hunk 1]
```diff
@@ -2,21 +2,30 @@
...
+
 public class Clanki {
-    private Ui ui;
-    private FlashcardList flashcardList;
-    private Parser parser;
+    private final Ui ui;
+    private final FlashcardList flashcardList;
+    private final Parser parser;
+    private ArrayList<Flashcard> flashcardsForToday;
+    private int currentFlashcardIndex;
```
Comment (joulev) [inline src/main/java/com/clanki/Clanki.java]: These variables are unused. I think it might be a good idea to just remove them, if we don't need them.

File: src/main/java/com/clanki/commands/ReviewCommand.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class ReviewCommand extends Command {
+    private static void reviewCurrentFlashcard(Ui display, Flashcard flashcard) {
+        boolean isCurrentCardPassed = false;
+        display.print("Let's see the next flashcard!");
+        while (!isCurrentCardPassed) {
+            display.print("Question for the current flashcard is: ", flashcard.getQuestion());
```
Comment (joulev) [inline src/main/java/com/clanki/commands/ReviewCommand.java]: The current test error is because there is a space at the end here, while there isn't in the `EXPECTED.txt` file.

File: src/main/java/com/clanki/commands/ReviewCommand.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            display.print("Are you ready to view the answer?");
+            String viewAnswerCommand = display.getUserCommand();
+            if (Objects.equals(viewAnswerCommand, "answer") || Objects.equals(viewAnswerCommand, "a")) {
```
Comment (joulev) [inline src/main/java/com/clanki/commands/ReviewCommand.java]: I think instead of requiring the user to input `answer` or `a` here, we can let them input anything, including an empty string. So we get the user input, but don't do anything with it.

File: src/main/java/com/clanki/commands/ReviewCommand.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                display.print(flashcard.getAnswer(), "Have you got this correct?");
+                String ifCorrectAnswerCommand = display.getUserCommand();
```
Comment (joulev) [inline src/main/java/com/clanki/commands/ReviewCommand.java]: I don't think this is a very good name for this variable. I think `answerIfCorrect` is a better name.

File: src/main/java/com/clanki/commands/ReviewCommand.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                if (Objects.equals(ifCorrectAnswerCommand, "yes") || Objects.equals(ifCorrectAnswerCommand, "y")) {
```
Comment (joulev) [inline src/main/java/com/clanki/commands/ReviewCommand.java]: I think this is buggy. [We should use something like `ifCorrectAnswerCommand.equals("yes")` instead](https://stackoverflow.com/q/13001971).

File: src/main/java/com/clanki/commands/ReviewCommand.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+import com.clanki.objects.Flashcard;
+import com.clanki.objects.FlashcardList;
+import com.clanki.ui.Ui;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class ReviewCommand extends Command {
+    private static void reviewCurrentFlashcard(Ui display, Flashcard flashcard) {
+        boolean isCurrentCardPassed = false;
```
Comment (joulev) [inline src/main/java/com/clanki/commands/ReviewCommand.java:11]: Since both branches of the correct/incorrect answers below end up assigning this variable to `true`, I'm a bit confused as in what this variable is for.

File: src/main/java/com/clanki/objects/FlashcardList.java [hunk 7]
```diff
@@ -3,7 +3,7 @@
 import java.util.ArrayList;
 
 public class FlashcardList {
-    private ArrayList<Flashcard> flashcards;
+    private final ArrayList<Flashcard> flashcards;
```
Comment (joulev) [inline src/main/java/com/clanki/objects/FlashcardList.java:6]: Why is `flashcards` a `final` variable?

File: src/main/java/com/clanki/ui/Ui.java [hunk 8]
```diff
@@ -46,4 +46,16 @@ public void printByeMessage() {
...
         System.out.println("The input is in an incorrect format.");
     }
+
+    /**
+     * reused from Kong Dehao ip
+     * for general printing to console
+     *
+     * @param message multiple strings to be shown to user
+     */
+    public void print(String... message) {
```
Comment (joulev) [inline src/main/java/com/clanki/ui/Ui.java]: I think a name like `printlnSeveralStrings` would be a better name. `print` can be mistook as printing something on the same line, and I think the `text-ui-test` bug above is due to this exact mistake.

File: text-ui-test/runtest.bat [hunk 9]
```diff
@@ -17,3 +17,4 @@ java -jar %jarloc% < ..\..\text-ui-test\input.txt > ..\..\text-ui-test\ACTUAL.TX
 cd ..\..\text-ui-test
 
 FC ACTUAL.TXT EXPECTED.TXT >NUL && ECHO Test passed! || Echo Test failed!
+FC ACTUAL.TXT EXPECTED.TXT
```
Comment (joulev) [inline text-ui-test/runtest.bat]: May I ask what is the purpose of this change?

**Review by joulev — CHANGES_REQUESTED** [review]

File: src/main/java/com/clanki/commands/ReviewCommand.java (hunk 6 above)
Comment (kdh3799) [reply src/main/java/com/clanki/commands/ReviewCommand.java:11]: When user gives a invalid input, it will stay false, so the flashcard will be shown again for user to review, instead of going to the next flashcard

File: src/main/java/com/clanki/objects/FlashcardList.java (hunk 7 above)
Comment (kdh3799) [reply src/main/java/com/clanki/objects/FlashcardList.java:6]: @SongZijin Hi, is this supposed to be final?

**Review by joulev — APPROVED** [review]
